old/lib/parser.py

```diff
@@ -370,14 +371,21 @@ def target():
 
     def kill_process(self, process):
         """Kill ``process`` and all its child processes."""
-        pid = process.pid
-        pids = [pid]
-        pids.extend(self.get_process_children(pid))
-        for pid in pids:
-            try:
-                os.kill(pid, SIGKILL)
-            except OSError:
-                pass
+        
+        # Obtain PID of Operation System.
+        osPID = os.getpid()
```

Comment:
I believe this is going to get the python process that is currently running. This will result in the program self-terminating, which is not what we want. Was there a reason that we did not continue to use `process.pid`?

Reply:
I don't believe we discussed whether or not to use `process.pid` for Windows. This is my bad. I took your code suggestion on Slack too literally.
`if platform.system() is not 'Windows':
     pid = process.pid
     pids = [pid]
     pids.extend(self.get_process_children(pid))
     for pid in pids:
           try:
                os.kill(pid, signal.SIGKILL)
            except OSError:
                 pass
        else:
              pid = process.pid
              pids = [pid]
              pids.extend(self.get_process_children(pid))
              for pid in pids:
                   try:
                       os.kill(pid, signal.SIGTERM)
                   except OSError:
                       pass`
Is probably what we actually want. We should wrap the logic executed in the conditional in a function and pass the OS specific signal function as an argument.